### PR TITLE
Make CommonDBRelations purgeable with the delete right

### PR DESCRIPTION
**Pull request by RomainLvr**
## Checklist before requesting a review

*Please delete options that are not relevant.*

- [x] I have read the CONTRIBUTING document.
- [x] I have performed a self-review of my code.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] This change requires a documentation update.

## Description

- It fixes !35411
- Here is a brief description of what this PR does

## Screenshots (if appropriate):

To remove a relationship between two items (e.g., `Domain` -> `Computer`), the system currently requires purge rights on the source item (`Domain` in this case). This behavior is problematic because granting purge rights not only allows the removal of relationships but also permits the deletion of items themselves. This creates a security concern when the intent is solely to manage relationships without affecting the items involved.

image

image

### CODE REVIEW

File: src/MassiveAction.php [hunk 1]
```diff
@@ -795,15 +795,17 @@ public static function getAllMassiveActions($item, $is_deleted = false, ?CommonD
...
                     $actions[$self_pref . 'purge'] = _x('button', 'Delete permanently');
                 }
                 if ($item instanceof CommonDropdown) {
                     $actions[$self_pref . 'purge_but_item_linked']
                      = _x('button', 'Delete permanently even if linked items');
                 }
+            } elseif ($candelete) {
+                if ($item instanceof CommonDBRelation) {
+                    $actions[$self_pref . 'delete'] = _x('button', 'Delete permanently the relation with selected elements');
+                }
```
Comment (stonebuzz) [inline src/MassiveAction.php:808]: Purging with only the deletion right would set a precedent in GLPI.

If I take the example of `Item_OperatingSystem` which is also a `CommonDBRelation`, the difference (in the code) is that `Item_OperatingSystem` does not depend on the` $rightname = "operatingsystem"`. 

It even specifies to `CommonDBRelation` not to check the rights on the `OperatingSystem` item.

`public static $checkItem_2_Rights = self::DONT_CHECK_ITEM_RIGHTS;`

This simple patch should be sufficient.

```patch
diff --git a/src/Domain_Item.php b/src/Domain_Item.php
index e349b077b6..c1ae17bcf1 100644
--- a/src/Domain_Item.php
+++ b/src/Domain_Item.php
@@ -42,7 +42,7 @@ class Domain_Item extends CommonDBRelation
     public static $itemtype_2 = 'itemtype';
     public static $items_id_2 = 'items_id';
 
-    public static $rightname = 'domain';
+    public static $checkItem_1_Rights = self::DONT_CHECK_ITEM_RIGHTS;
 
     public static function getTypeName($nb = 0)
     {
```

This allows the `purge` of the relation without granting the purge right on `Domain`.

`CommonDBRelation` will verify that the user has the right to `update` the main element, in this case, `Computer`, to allow the purge of the relation.

**Review by stonebuzz — CHANGES_REQUESTED** [review]

**Comment by cconard96** [comment]
I'm not sure this makes sense. In fact, it may only require delete permissions for Domains because `Domain_Item` specifies that as its rightname. In general, deleting a relationship only requires UPDATE rights for both of the associated items (Domain and Asset in this case).

**Comment by stonebuzz** [comment]
> I'm not sure this makes sense. In fact, it may only require delete permissions for Domains because `Domain_Item` specifies that as its rightname. In general, deleting a relationship only requires UPDATE rights for both of the associated items (Domain and Asset in this case).

So only this should enought

```patch
diff --git a/src/Domain_Item.php b/src/Domain_Item.php
index e349b077b6..9be6dcc3ae 100644
--- a/src/Domain_Item.php
+++ b/src/Domain_Item.php
@@ -42,8 +42,6 @@ class Domain_Item extends CommonDBRelation
     public static $itemtype_2 = 'itemtype';
     public static $items_id_2 = 'items_id';
 
-    public static $rightname = 'domain';
-
     public static function getTypeName($nb = 0)
     {
         return _n('Domain item', 'Domain items', $nb);

```
In this way, `CommonDBRelation` check the `update` permissions on both sides.

(and it works)

**Review by cedric-anne — COMMENTED** [review]
Fix seems OK. Should it target the 10.0 branch?

**Comment by RomainLvr** [comment]
> Fix seems OK. Should it target the 10.0 branch?

This fix does not work in 10.0

**Comment by trasher** [comment]
Please fix tests

File: templates/generic_show_form.html.twig [hunk 2]
```diff
@@ -479,7 +479,7 @@
                      {% endif %}
 
                      {% if item.isField('sysdescr') %}
-                        {{ fields.textareaField('sysdescr', item.fields['sysdescr'], __('Sysdescr')) }}
+                        {{ fields.textareaField('sysdescr', item.fields['sysdescr'], __('Sysdescr'), field_options) }}
```
Comment (cedric-anne) [inline templates/generic_show_form.html.twig:482]: The related commit should not be present in the current PR.

**Comment by RomainLvr** [comment]
PR redone here  https://github.com/glpi-project/glpi/pull/18553

**Pull request closed without merging**